src/rust/engine/async_semaphore/src/lib.rs

```diff
@@ -29,29 +29,19 @@
 
 use std::collections::VecDeque;
 use std::future::Future;
-use std::pin::Pin;
 use std::sync::Arc;
-use std::task::{Context, Poll, Waker};
 
 use parking_lot::Mutex;
-
-struct Waiter {
-  id: usize,
-  waker: Waker,
-}
+use tokio::sync::{Semaphore, SemaphorePermit};
 
 struct Inner {
-  waiters: VecDeque<Waiter>,
-  available_ids: VecDeque<usize>,
-  available_permits: usize,
-  // Used as the source of id in Waiters's because
-  // it is monotonically increasing, and only incremented under the mutex lock.
-  next_waiter_id: usize,
+  sema: Semaphore,
```

Comment:
Nit: I'd recommend spelling out the whole thing so that there's no ambiguity.

Reply:
Would agree in general, but it should be pretty clear what this is given the context.